### PR TITLE
Normalize tag support

### PR DESCRIPTION
**Pull request by gnydick**
This new plugin type allows you to provide a plugin to normalize tags. My use case for implementing this is the fact that the slew of Cloud Native systems coming online all export their metrics slightly differently and opentsdb doesn't support fuzzy matches or specifying which tags are equivalent to each other.

I was receiving container, container-name, container_name, etc., etc., etc.

This plugin type requires a normalizeTags(tags) method. The tags are passed in and a normalized set are returned.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # along with this library.  If not, see <http://www.gnu.org/licenses/>.
 
 # Semantic Versioning (see http://semver.org/).
-AC_INIT([opentsdb], [2.4.0], [opentsdb@googlegroups.com])
+AC_INIT([opentsdb], [2.4.0-tnp], [opentsdb@googlegroups.com])
```
Comment (johann8384) [inline configure.ac]: We probably don't need this change.

File: src/core/TSDB.java [hunk 2]
```diff
@@ -12,59 +12,17 @@
 // see <http://www.gnu.org/licenses/>.
 package net.opentsdb.core;
 
-import java.io.File;
```
Comment (johann8384) [inline src/core/TSDB.java]: So, it looks like you did the "auto format" in IntelliJ, I've been tempted to commit this myself. Would it be possible to isolate just the changes relevant to your new plugin type?

I love the concept and would be happy to get this merged if we can remove the unrelated changes.

File: src/tsd/RTPublisher.java [hunk 3]
```diff
@@ -28,7 +28,7 @@
  * after they are queued for storage. In the future we may support publishing
  * meta data or other types of information as changes are made.
  * <p>
- * <b>Note:</b> Implementations must have a parameterless constructor. The
```
Comment (johann8384) [inline src/tsd/RTPublisher.java]: You can leave in all the "trailing whitespace" fixes if you want! :)

**Review by johann8384 — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 1 above)
Comment (gnydick) [reply configure.ac]: Sounds good!

**Comment by johann8384** [comment]
Merging in #2302. @gnydick I know it's been, um, years, but if you have an example of a working tag normalization plugin that would be great!

**Review by sins921 — APPROVED** [review]

**Comment by gnydick** [comment]
Let me see if I have an example for you...

**Comment by gnydick** [comment]
Here's an example plugin.  https://github.com/OpenTSDB/opentsdb/commit/6aba0eda2dcf3dea1f390b2484cd00915e649abc

**Review by sins921 — APPROVED** [review]

**Pull request closed without merging**